mir-module/src/main/java/org/mycore/mir/authorization/accesskeys/backend/MIRAccessKey.java

```diff
@@ -0,0 +1,204 @@
+
+import org.mycore.datamodel.metadata.MCRObjectID;
+
+@NamedQueries({
+    @NamedQuery(name = "MIRAccessKey.getById",
+        query = "SELECT k"
+            + "  FROM MIRAccessKey k"
+            + "  WHERE k.objectIdString = :objId"),
+    @NamedQuery(name = "MIRAccessKey.getByValue",
+        query = "SELECT k"
+            + "  FROM MIRAccessKey k"
+            + "  WHERE k.value = :value AND k.objectIdString = :objId"),
+    @NamedQuery(name = "MIRAccessKey.clearById",
+        query = "DELETE"
+            + "  FROM MIRAccessKey k"
+            + "  WHERE k.objectIdString = :objId"),
+    @NamedQuery(name = "MIRAccessKey.clear",
+        query = "DELETE"
+            + "  FROM MIRAccessKey k"),
+})
+ 
+/**
+ * Access keys for a {@link MCRObject}.
+ * An access keys contains a value and a type.
+ * Value is the key value of the key and type the permission.
+ */
+@Entity
+@Table(name = "miraccesskey")
+public class MIRAccessKey {
+
+    private static final long serialVersionUID = 1L;
+
+    /** The unique and internal information id */
+    private UUID id;
+
+    /** The access key information*/
+    private MCRObjectID mcrObjectId; 
+
+    /** The key value*/
+    private String value;
+
+    /** The permission type*/
+    private String type;
+
+    private MIRAccessKey() {
+    }
+
+    /**
+     * Creates a new access key with value and type.
+     *
+     * @param value the value the user must know to acquire permission.
+     * @param type the type of permission.
+     */
+    public MIRAccessKey(final String value, final String type) {
+        setValue(value);
+        setType(type);
+    }
+
+    /**
+     * Creates a new access key with value and type.
+     *
+     * @param objectId the assigned {@link MCRObjectID}.
+     * @param value the value the user must know to acquire permission.
+     * @param type the type of permission.
+     */
+    public MIRAccessKey(final MCRObjectID objectId, final String value, final String type) {
+        setObjectId(objectId);
+        setValue(value);
+        setType(type);
+    }
+
+    /**
+     * @return the linked mcrObjectId
+     */
+    @JsonIgnore
+    @Transient
+    public MCRObjectID getObjectId() {
+        return mcrObjectId;
+    }
+
+    /**
+     * @param mcrObjectId the {@link MCRObjectID} to set
+     */
+    public void setObjectId(final MCRObjectID mcrObjectId) {
+        this.mcrObjectId = mcrObjectId;
+    }
+
+    /**
+     * @return objectId as String
+     */
+    @JsonIgnore
+    @Column(name = "objectid", 
+        nullable = false)
+    public String getObjectIdString() {
+        return mcrObjectId.toString();
+    }
+
+    /**
+     * @param objectIdString id as String
+     */
+    public void setObjectIdString(String objectIdString) {
+        this.mcrObjectId = MCRObjectID.getInstance(objectIdString);
+    }
+
+    /**
+     * @return internal id
+     */
+    @Id
+    @GeneratedValue
+    @Column(name = "accesskey_id",
+        nullable = false)
```

Comment:
use "int" type:
```
@Id
@GeneratedValue(strategy = GenerationType.IDENTITY)
public int getInternalId()
```

Reply:
`GenerationType.IDENTITY` is not compatible with UUID. That is why I use the generic approach with the default strategy `GenerationType.AUTO`. I could specify a UUID generator instead of the generic approach and use that as a strategy.

Reply:
We should implement this consistently. Type "int" is used in all other tables ...

Reply:
I would like to manage the keys later via REST. I am not a friend of giving out ids, so I chose UUID as identifier for security reasons.

Reply:
There are artificial and natural ids. You can keep MCRObjectID, type and AccessToken for Rest-API, while hiding the internal ID completely.